src/shared/extensionUtilities.ts

```diff
@@ -109,6 +132,14 @@ export function isCloud9(flavor: 'classic' | 'codecatalyst' | 'any' = 'any'): bo
     return (flavor === 'classic' && !codecat) || (flavor === 'codecatalyst' && codecat)
 }
 
+export function isSageMaker(): boolean {
+    return vscode.env.appName === sageMakerAppname
+}
+
+export function hasVendedIamCredentials(isC9: boolean = isCloud9(), isSM: boolean = isSageMaker()) {
```

Comment:
a docstring can provide some useful insight
```suggestion
/** Credentials are placed in ~/.aws/ by the environment. */
export function hasVendedIamCredentials(isC9: boolean = isCloud9(), isSM: boolean = isSageMaker()) {
```

Reply:
Note that calling isCloud9() etc in the default parameter values caused that function to be executed at module-load time. That can be problematic, especially for this module which is a low-level module used to initialize early parts of the app. 

To be defensive, instead of assigning default parameter values, make them optional and then check for undefined in the function body.

```
isC9 ??= isCloud()
...
```